src/Result/Ok.php

```diff
@@ -54,185 +54,150 @@ public function __construct($value, ...$pass)
 
         if (!($result instanceof Result)) {
-            throw new ResultException("Op must return a Result");
+            throw new ResultException('Op must return a Result');
         }
 
         return $result;
     }
 
     /**
-     * Returns res if the result is Err, otherwise returns the Ok value of self.
-     *
-     * @param Result $res
-     * @return Result
+     * @inheritDoc
      */
     public function or(Result $res): Result
     {
         return $this;
     }
 
     /**
-     * Calls op if the result is Err, otherwise returns the Ok value of self.
-     *
-     * @param Closure $op
-     * @return Result
+     * @inheritDoc
      */
     public function orElse(Closure $op): Result
     {
         return $this;
     }
 
     /**
-     * Unwraps a result, yielding the content of an Ok. Else, it returns optb.
+     * @inheritDoc
      *
-     * @param $optb
-     * @return mixed
-     * @psalm-return T|mixed
+     * @psalm-return T
      */
     public function unwrapOr($optb)
     {
         return $this->value;
     }
 
     /**
-     * Unwraps a result, yielding the content of an Ok. If the value is an Err then it calls op with its value.
+     * @inheritDoc
      *
-     * @param Closure $op
-     * @return mixed
-     * @psalm-return T|mixed
+     * @psalm-return T
      */
     public function unwrapOrElse(Closure $op)
     {
         return $this->value;
     }
 
     /**
-     * Unwraps a result, yielding the content of an Ok.
-     *
-     * @throws if the value is an Err.
-     * @return mixed
-     * @psalm-return T
+     * @inheritDoc
      */
     public function unwrap()
     {
         return $this->value;
     }
 
     /**
-     * Unwraps a result, yielding the content of an Ok.
-     *
-     * @throws Exception (the message) if the value is an Err.
-     * @param Exception $msg
-     * @return mixed
-     * @psalm-return T
+     * @inheritDoc
      */
     public function expect(Exception $msg)
     {
         return $this->value;
     }
 
     /**
-     * Unwraps a result, yielding the content of an Err.
+     * @inheritDoc
      *
+     * @psalm-return never-return
      * @throws ResultException if the value is an Ok.
-     * @return mixed
-     * @psalm-return E
      */
     public function unwrapErr()
     {
-        throw new ResultException("Unwrapped with the expecation of Err, but found Ok");
+        throw new ResultException('Unwrapped with the expectation of Err, but found Ok');
     }
 
     /**
-     * Applies values inside the given Results to the function in this Result.
+     * @inheritDoc
      *
-     * @param Result[] ...$inArgs Results to apply the function to.
-     * @psalm-param Result ...$inArgs
-     * @return Result
      * @throws ResultException
-     * @psalm-suppress MissingClosureParamType
+     *
+     * @psalm-return Result<array,E>
+     *
+//     * @psalm-assert T!=callable $this->value
+//     * @psalm-assert T!=callable(T=,...mixed=):Result<array,E> $this->value
```

Comment:
Is the double comment a mistake?

Reply:
I didn't figure out Psalm concept for `psalm-assert` annotations yet. 
I tried to check the difference "with" and  "without" this annotation. 
Link for docs: https://github.com/vimeo/psalm/blob/master/docs/supported_annotations.md#psalm-assert-psalm-assert-if-true-and-psalm-assert-if-false

Will be be dropped or fixed in the final implementation.